apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/SolrClientInterceptor.java

```diff
@@ -0,0 +1,185 @@
+                if (update instanceof UpdateRequest) {
+                    UpdateRequest ur = (UpdateRequest) update;
+                    List<SolrInputDocument> documents = ur.getDocuments();
+                    if (documents == null) {
+                        actionName = "DELETE_BY_IDS";
+
+                        List<String> deleteBy = ur.getDeleteById();
+                        if (deleteBy == null) {
+                            actionName = "DELETE_BY_QUERY";
+                            deleteBy = ur.getDeleteQuery();
+                        }
+                        if (deleteBy != null) {
+                            span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+                            span.tag(SolrjTags.TAG_DELETE_VALUE, deleteBy.toString());
+                        }
+                    } else {
+                        span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+                        span.tag(SolrjTags.TAG_DOCS_SIZE, String.valueOf(documents.size()));
+                        span.tag(SolrjTags.TAG_COMMIT_WITHIN, String.valueOf(update.getCommitWithin()));
+                    }
+                } else {
+                    span = getSpan(collection, request.getPath(), "", instance.getRemotePeer());
+                }
+            } else {
+                actionName = action.name();
+                if (action == AbstractUpdateRequest.ACTION.COMMIT) {
+                    span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+                    span.tag(SolrjTags.TAG_SOFT_COMMIT, params.get(UpdateParams.SOFT_COMMIT, ""));
+                } else {
+                    span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+                    span.tag(SolrjTags.TAG_MAX_OPTIMIZE_SEGMENTS, params.get(UpdateParams.MAX_OPTIMIZE_SEGMENTS, "1"));
+                }
+            }
+        } else if (request instanceof QueryRequest) {
+            String operatorName = String.format("solrJ/%s%s", collection, request.getPath());
+            span = ContextManager.createExitSpan(operatorName, instance.getRemotePeer())
+                    .setComponent(ComponentsDefine.SOLRJ)
+                    .setLayer(SpanLayer.DB);
+
+            span.tag(SolrjTags.TAG_SORT_BY, params.get(CommonParams.SORT, ""));
+            span.tag(SolrjTags.TAG_START, params.get(CommonParams.START, "0"));
+            span.tag(SolrjTags.TAG_QT, params.get(CommonParams.QT, request.getPath()));
+        }
+
+        if (null == span)
+            return;
+
+        span.tag(SolrjTags.TAG_COLLECTION, collection);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                              Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        if (!ContextManager.isActive()) return ret;
```

Comment:
i think this maybe a problem
the reason is this span is a exitSpan and in beforeMethod span may be null and u do return
but in afterMethod `ContextManager.activeSpan()` could get a entry span which is not solr span sometimes.
please consider this

Reply:
Why there is a context existing in after, but not in start? Could you show me some cases? I am not following.

Reply:
Are you saying read context doesn't belong to Solr plugin?

Reply:
I have  two ways to resolve it. Could you help me to choose?
1. Using the same conditions with beforeMethod.
```java
        SolrRequest<?> request = (SolrRequest<?>) allArguments[0];
        if (!("".equals(request.getPath()) || request instanceof QueryRequest)) {
            return ret;
        }
```
2. Put a flag into RuntimeContext.
```java
        if (ContextManager.getRuntimeContext().get("solrj_mark") == null) {
            return ret;
        }
```

Reply:
I prefer (1)